app/frontend/src/features/auth/authActions.ts

```diff
@@ -2,20 +2,12 @@ import { createAsyncThunk } from "@reduxjs/toolkit";
 import { service } from "../../service";
 import { SignupArguments } from "../../service/user";
 
-const {
-  completeSignup,
-  getCurrentUser,
-  getUser,
-  passwordLogin: apiPasswordLogin,
-  tokenLogin: apiTokenLogin,
-} = service.user;
```

Comment:
I think the solution I proposed (about making the service layer ignorant about the model and view) two weeks ago was better. I did that exactly for situations like this. The bad thing about your solution is that this problem will appear more often in the future and that production code is changed for testing purposes.

Reply:
If the circular dependencies are this flaky I'm beginning to think we should just do the `setToken` hack and never have to worry about it again.

Reply:
Do we use the token anywhere else in the app apart from the client needing it? We might be better off moving it out of the store and into something even more global like local storage?

A separate issue, but this has got me thinking that the way we do auth in our app is not that secure, as we're essentially using the token as a session token but it's stored in local storage (with Redux persist) which is known to be vulnerable to XSS. This is [one of](https://developer.okta.com/blog/2018/06/20/what-happens-if-your-jwt-is-stolen) the many nice articles that explains the issue. It mentions JWT but it's not that different from what we're doing right now unless I misunderstood something?

Probably one for a wider architectural discussion though rather than changing it here!

Reply:
> Do we use the token anywhere else in the app apart from the client needing it?

Nope, just the interceptors and a few of the API functions need it.

> We might be better off moving it out of the store and into something even more global like local storage?

Seems over-complicated to me. 

I don't know about Local Storage and XSS, but it's just a session token (more like a cookie than a JWT) and the server can invalidate whenever it so the JWT problems don't apply as far as I can tell.

Reply:
> Seems over-complicated to me.

Is it that complicated though if all it takes is `localStorage.getItem('token')` and `localStorage.setItem('token', <value>)`? I'd say simpler and less Redux code too as a result, if security turned out to be not a concern.

Reply:
Instead of a `setToken` hack we could also initialize the service with a getToken function.

Reply:
instead of `localStorage` we better use `indexedDB`

Reply:
You are right: the token can be used that way. For instance by someone using the same browser later.

Reply:
But for user friendliness it is nice if the user is given the choice to stay logged in.

Reply:
> instead of `localStorage` we better use `indexedDB`

What would be the benefit of using `indexedDB` if we are not storing a huge amount of data in this case?

Reply:
>  Once the user has the token, they're basically free to request as the authenticated user right?

Yes but that's also true for cookies or any system which will keep you logged in.

How about storing the session token in a same site only cookie?

Reply:
The main difference with cookies is you can have HTTP only cookies, which means JS won't be able to read from it and that eliminates the potential XSS problem. Together with the same site only cookie you mentioned and make sure we only support modern browsers, sounds good to me.

This would require the backend to change to read/set the cookies instead of reading from the auth header though, so maybe best to do it in a separate PR?